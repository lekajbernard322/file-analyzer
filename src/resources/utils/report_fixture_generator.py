from os import listdir
from os.path import isfile, join
import json
from datetime import datetime, timezone
from bson.objectid import ObjectId

INPUT_DIR = r'../report_data/'
OUTPUT_PATH = r'../../backend/app/report/fixtures/report_fixture.json'

file_paths = [(join(INPUT_DIR, file_name), file_name)
              for file_name in listdir(INPUT_DIR)
              if isfile(join(INPUT_DIR, file_name))]


def prepare_output_data(i, file_name, content):
    return {
        'model': 'report.report',
        'pk': str(ObjectId()),
        'fields': {
            'name': file_name,
            'content': content,
            'created': datetime.now(tz=timezone.utc).isoformat()
        }
    }


data = []
for index, (input_file_path, input_file_name) in enumerate(file_paths):
    print(f'[*] Processing file: {input_file_path}')

    with open(input_file_path, 'r') as input_file:
        json_content = json.load(input_file)

        output_data = prepare_output_data(index, input_file_name, json_content)

        data.append(output_data)

print(f'[*] Dumping to fixture file: {OUTPUT_PATH}')
with open(OUTPUT_PATH, 'w', encoding='utf-8') as output_file:
    json.dump(data, output_file, indent=4)

print(f'[\u2714] Done')
