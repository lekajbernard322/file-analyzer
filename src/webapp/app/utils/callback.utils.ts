export const createOnPressEnter = (callback) => ({key}) => key === 'Enter' && callback && callback()
