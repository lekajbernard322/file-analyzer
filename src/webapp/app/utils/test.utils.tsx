import thunk from "redux-thunk";
import {createMemoryHistory} from "history";
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import React from "react";
import configureStore from "redux-mock-store";

export const renderWithContext = (additionalState, ui) => {
    const mockStore = configureStore([thunk]);
    const store = mockStore(additionalState);
    const history = createMemoryHistory();

    const Wrapper = ({children}) => (
        <Provider store={store}>
            <Router history={history}>
                {children}
            </Router>
        </Provider>);

    return {rendered: render(ui, {wrapper: Wrapper}), store, history};
};
