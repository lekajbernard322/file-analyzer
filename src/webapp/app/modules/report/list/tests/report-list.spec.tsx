import React from "react";
import { fireEvent } from "@testing-library/react";
import ReportListPage from "../report-list";
import {
  getDllStatistics,
  getFunctionStatistics,
  getReports,
  clearReportList,
} from "../report-list.reducer";
import {renderWithContext} from "../../../../utils/test.utils";

export const mockData = {
  report: {
    list: {
      data: [
        {
          _id: "1",
          name: "report1",
          created: "some date",
        },
        {
          _id: "2",
          name: "report1",
          created: "some date",
        },
      ],
      total: 3,
      functionStatistics: {
        averageNumberOfFunctions: 1,
        minNumberOfFunctions: 2,
        maxNumberOfFunctions: 3,
      },
      dllStatistics: {
        averageNumberOfDlls: 4,
        minNumberOfDlls: 5,
        maxNumberOfDlls: 6,
      },
      loading: false,
      error: false,
    },
  },
};

describe("Report", () => {
  const renderComponent = () => {
    return renderWithContext(mockData, <ReportListPage />);
  };

  it("should render correctly", () => {
    const {
      rendered: { container },
    } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it("should execute fetches for data", () => {
    const { store } = renderComponent();

    const [
      clearReportListAction,
      getReportsPendingAction,
      getFunctionStatisticsPendingAction,
      getDllStatisticsPendingAction,
    ] = store.getActions();

    expect(clearReportListAction).toMatchObject({ type: clearReportList.type });
    expect(getReportsPendingAction).toMatchObject({
      type: getReports.pending.type,
    });
    expect(getFunctionStatisticsPendingAction).toMatchObject({
      type: getFunctionStatistics.pending.type,
    });
    expect(getDllStatisticsPendingAction).toMatchObject({
      type: getDllStatistics.pending.type,
    });
  });

  it("should search if clicked on filter button", () => {
    const {
      rendered: { queryByPlaceholderText, queryByText },
      store,
    } = renderComponent();

    const searchInput = queryByPlaceholderText("Search");
    const filterButton = queryByText("Filter");

    expect(searchInput).toBeInTheDocument();
    expect(filterButton).toBeInTheDocument();

    fireEvent.change(searchInput, { target: { value: "test" } });
    fireEvent.click(filterButton);

    const lastAction = store.getActions().slice(-1)[0];

    expect(lastAction).toMatchObject({
      meta: {
        arg: {
          limit: 2,
          offset: 0,
          search: "test",
        },
      },
      type: "reportList/get_reports/pending",
    });
  });

  it("should load more when clicked on load more button", () => {
    const {
      rendered: { queryByText },
      store,
    } = renderComponent();

    const loadMoreButton = queryByText("Load more");

    expect(loadMoreButton).toBeInTheDocument();

    fireEvent.click(loadMoreButton);

    const lastAction = store.getActions().slice(-1)[0];

    expect(lastAction).toMatchObject({
      meta: {
        arg: {
          limit: 2,
          offset: 2,
          search: "",
        },
      },
      type: "reportList/get_reports/pending",
    });
  });

  it("should navigate to report details when clicked on view button", () => {
    const {
      rendered: { queryAllByText },
      history,
    } = renderComponent();

    const [firstViewButton] = queryAllByText("View");

    expect(firstViewButton).toBeInTheDocument();

    fireEvent.click(firstViewButton);

    expect(history.location.pathname).toEqual("/details/1");
  });
});
