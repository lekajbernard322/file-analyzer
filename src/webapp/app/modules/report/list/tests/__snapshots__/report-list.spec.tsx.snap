// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report should render correctly 1`] = `
<div>
  <div>
    <div
      class="breadcrumb"
    >
      <a
        href="/"
      >
        Reports
        /
      </a>
    </div>
    <h1
      class="headline"
    >
      PE metadata reports
    </h1>
    <h3
      class="subheading"
    >
      List of available reports. Click on a report to view more details about it.
    </h3>
    <div>
      <hr />
    </div>
    <div
      class="horizontal-container"
    >
      <div
        class="information-highlighter"
      >
        <div
          class="information-highlighter-message"
        >
          Total number of reports
        </div>
        <div
          class="information-highlighter-value"
        >
          3
        </div>
      </div>
      <div
        class="information-highlighter"
      >
        <div
          class="information-highlighter-message"
        >
          Average number of DLLs per report
        </div>
        <div
          class="information-highlighter-value"
        >
          1
        </div>
      </div>
      <div
        class="information-highlighter"
      >
        <div
          class="information-highlighter-message"
        >
          Average number of functions per report
        </div>
        <div
          class="information-highlighter-value"
        >
          4
        </div>
      </div>
    </div>
    <div>
      <hr />
    </div>
    <input
      placeholder="Search"
      value=""
    />
    <button>
      Filter
    </button>
    <div
      class="page-content"
    >
      <div>
        <table>
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                Created
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                report1
              </td>
              <td>
                some date
              </td>
              <td
                class="actions"
              >
                <a
                  href="/details/1"
                >
                  View
                </a>
              </td>
            </tr>
            <tr>
              <td>
                report1
              </td>
              <td>
                some date
              </td>
              <td
                class="actions"
              >
                <a
                  href="/details/2"
                >
                  View
                </a>
              </td>
            </tr>
          </tbody>
        </table>
        <button>
          Load more
        </button>
      </div>
    </div>
  </div>
</div>
`;
